pkg/metrics/resource.go

```diff
@@ -51,6 +52,12 @@ var (
 		Name: policyPreemptionMetricsName,
 		Help: "Number of preemption for the resource template. By the result, 'error' means a resource template failed to be preempted by other propagation policies. Otherwise 'success'.",
 	}, []string{"result"})
+
+	metricsAdapterPullDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    metricsAdapterPullDurationMetricsName,
+		Help:    "Duration in seconds taken by the karmada-metrics-adapter to pull metrics. By the result, 'error' means the metrics adapter failed to pull the metrics. Otherwise 'success'.",
+		Buckets: prometheus.ExponentialBuckets(0.001, 2, 12),
+	}, []string{"result"})
```

Comment:
Can we record the cluster name so that we can identify which cluster is causing the delay?

Reply:
Sure